src/main/scala/com/redislabs/provider/redis/rdd/RedisRDD.scala

```diff
@@ -29,48 +34,34 @@ class RedisKVRDD(prev: RDD[String],
     rddType match {
       case "kv"   => getKV(nodes, keys)
       case "hash" => getHASH(nodes, keys)
-      case "zset" => getZSET(nodes, keys)
     }
   }
 
   def getKV(nodes: Array[RedisNode], keys: Iterator[String]): Iterator[(String, String)] = {
     groupKeysByNode(nodes, keys).flatMap {
       x =>
-        {
-          val conn = x._1.endpoint.connect()
-          val stringKeys = filterKeysByType(conn, x._2, "string")
-          val pipeline = conn.pipelined
-          stringKeys.foreach(pipeline.get)
-          val res = stringKeys.zip(pipeline.syncAndReturnAll).iterator.
-            asInstanceOf[Iterator[(String, String)]]
-          conn.close
-          res
-        }
+      {
+        val conn = x._1.endpoint.connect()
+        val stringKeys = filterKeysByType(conn, x._2, "string")
+        val pipeline = conn.pipelined
+        stringKeys.foreach(pipeline.get)
+        val res = stringKeys.zip(pipeline.syncAndReturnAll).iterator.
+          asInstanceOf[Iterator[(String, String)]]
+        conn.close
+        res
+      }
     }.iterator
   }
   def getHASH(nodes: Array[RedisNode], keys: Iterator[String]): Iterator[(String, String)] = {
     groupKeysByNode(nodes, keys).flatMap {
       x =>
-        {
-          val conn = x._1.endpoint.connect()
-          val hashKeys = filterKeysByType(conn, x._2, "hash")
-          val res = hashKeys.flatMap(conn.hgetAll).iterator
-          conn.close
-          res
-        }
-    }.iterator
-  }
-  def getZSET(nodes: Array[RedisNode], keys: Iterator[String]): Iterator[(String, String)] = {
-    groupKeysByNode(nodes, keys).flatMap {
-      x =>
-        {
-          val conn = x._1.endpoint.connect()
-          val zsetKeys = filterKeysByType(conn, x._2, "zset")
-          val res = zsetKeys.flatMap(k => conn.zrangeWithScores(k, 0, -1)).
-            map(tup => (tup.getElement, tup.getScore.toString)).iterator
-          conn.close
-          res
-        }
+      {
+        val conn = x._1.endpoint.connect()
+        val hashKeys = filterKeysByType(conn, x._2, "hash")
```

Comment:
do we really need this filtering? I know it's not new but why? It's better to fail if the user sent non-hash ids